content/en/docs/concepts/overview/object-management-kubectl/kustomization.md

```diff
@@ -0,0 +1,752 @@
+at <<EOF > base/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-nginx
+spec:
+  selector:
+    matchLabels:
+      run: my-nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - name: my-nginx
+        image: nginx
+        command: ["start", "--host", "\$(MY_SERVICE_NAME)"]
+EOF
+
+# Create a base/service.yaml file
+cat <<EOF > base/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-nginx
+  labels:
+    run: my-nginx
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: my-nginx
+EOF
+# Create a base/kustomization.yaml
+cat <<EOF > base/kustomization.yaml
+resources:
+- deployment.yaml
+- service.yaml
+```
+This base can be used in multiple overlays. You can add different namePrefix or other cross-cutting fields
+in different overlays. Here are two overlays using the same base.
+```shell
+mkdir dev
+cat <<EOF > dev/kustomization.yaml
+bases:
+- ../base
+namePrefix: dev-
+EOF
+
+mkdir prod
+cat <<EOF > prod/kustomization.yaml
+bases:
+- ../base
+namePrefix: prod-
+EOF
+```
+
+## How to apply/view/delete objects from Kustomization
+Use `--kustomize` or `-k` in kubectl commands to recognize Resources managed by `kustomization.yaml`. 
+Note that `-k` should point to a kustomization directory, such as
+
+```shell
+kubectl apply -k <kustomization directory>/
+```
+Given following `kustomization.yaml` file and associated Resource files,
+```shell
+# Create a deployment.yaml file
+cat <<EOF > deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-nginx
+spec:
+  selector:
+    matchLabels:
+      run: my-nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        run: my-nginx
+    spec:
+      containers:
+      - name: my-nginx
+        image: nginx
+        ports:
+        - containerPort: 80
+EOF
+
+# Create a kustomization.yaml
+cat <<EOF >./kustomization.yaml
+namePrefix: dev-
+commonLabels:
+  app: my-nginx
+resources:
+- deployment.yaml
+EOF
+```
+Running following command will apply the Deployment object `dev-my-nginx`
+```shell
+> kubectl apply -k ./
+deployment.apps/dev-my-nginx created
+```
+Run following command to view the Deployment object
+```shell
+kubectl get -k ./
+```
+or 
+```shell
+kubectl describe -k ./
+```
+Run following command to delete the Deployment object
+```shell
+> kubectl delete -k ./
+deployment.apps "dev-my-nginx" deleted
+```
+
+
+## Kustomize Feature List
+Here is a list of all the features in Kustomize.
+
+| Field                 | Type                                                                                                         | Explanation                                                                        |  
+|-----------------------|--------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------|
```

Comment:
nit: This table does not render in preview. There are four extra spaces after the last pipe in the header row.
See pipe on right hand side of Explanation.

Reply:
updated